Client/Frontend/Browser/TabManager.swift

```diff
@@ -449,16 +449,20 @@ class TabManager: NSObject {
         guard let tempTabs = self.tempTabs where tempTabs.count ?? 0 > 0 else {
             return
         }
-        
         let tabsCopy = normalTabs
         restoreTabs(tempTabs)
+        self.isRestoring = true
         for tab in tempTabs {
             tab.showContent(true)
         }
         if !tempTabs[0].isPrivate ?? true {
             removeTabs(tabsCopy)
         }
         selectTab(tempTabs.first)
+        self.isRestoring = false
```

Comment:
But when undoing tabs we do use `restoreTabs` to do it. So in that case undoing tabs is the same as restoring tabs. The only reason I have to do isRestoring here is because undoing tabs also deletes any existing tabs and reselects a new tab. 

So what ends up happening is some of the animations dont happen (isRestoring is true) but then the `removeTab` on L:452 happens and I try animating that change. But now we are on a bad state and we crash 😢 

By wrapping everything in a restoring state and calling `didFinishRestore` at the end I can safely reload without crashing. Hopefully 🤞